src/game_renderer/GameRenderer.h

```diff
@@ -3,16 +3,39 @@
 #ifndef GAME_RENDERER_H_
 #define GAME_RENDERER_H_
 
+#include <cmath>
+#include <fstream>
+#include <glm/gtc/type_ptr.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+#include <glob.h>
+#include <iostream>
+#include <unordered_map>
 #include "game_state/GameState.h"
+#include "GLFW/glfw3.h"
+#include "GL/glew.h"
+#include "GLSL.h"
+#include "json.hpp"
+#include "MatrixStack.h"
+#include "Program.h"
+#include "Texture.h"
 
 class GameRenderer {
  public:
   GameRenderer();
   ~GameRenderer();
-
+  GLFWwindow* GetWindow();
+  std::shared_ptr<Program> ProgramFromJSON(std::string filepath);
+  std::vector<std::string> ListFiles(const std::string& pattern);
```

Comment:
These two new functions shouldn't be publicly exposed since it appears that they are only used in GameRenderer.cpp. Even better, you could try putting them in the anonymous namespace of GameRenderer.cpp so they don't have to go in this header at all. To do this, write the two functions in a "namespace { ... }" in GameRenderer.cpp and remove the "GameRenderer::" from the beginning of the method signature.